src/main/kotlin/io/snyk/plugin/ui/ReferenceChooserComboboxDialog.kt

```diff
@@ -97,16 +97,13 @@ class ReferenceChooserDialog(val project: Project) : DialogWrapper(true) {
             val baseBranch = getSelectedItem(it.value) ?: ""
             val referenceFolderControl = referenceFolders[folderConfig]
             val referenceFolder = referenceFolderControl?.text ?: ""
-            if (baseBranch.isNotBlank()) {
-                folderConfigSettings.addFolderConfig(folderConfig.copy(baseBranch = baseBranch))
-            }
-            if (referenceFolder.isNotBlank()) {
-                folderConfigSettings.addFolderConfig(folderConfig.copy(referenceFolderPath = referenceFolder))
+            if (referenceFolder.isNotBlank() || baseBranch.isNotBlank()) {
+                folderConfigSettings.addFolderConfig(folderConfig.copy(baseBranch = baseBranch, referenceFolderPath = referenceFolder))
             }
         }
 
         if (doValidate() == null) {
-            runInBackground("Snyk: updating configuration") {
+            runAsync {
```

Comment:
Hm... this runs without progress indication. Also, I don't understand why it would block the UI thread?!